### PR TITLE
Fix `Node::physical_rect` and add a `physical_size` method

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

* `Node::physical_rect` divides the logical size of the node by the scale factor, when it should multiply.
* Add a `physical_size` method to `Node` that calculates the physical size of a node.

---

## Changelog

* Added a method `physical_size` to `Node` that calculates the physical size of the `Node` based on the given scale factor.
* Fixed the `Node::physical_rect` method, the logical size should be multiplied by the scale factor to get the physical size.
* Removed the `scale_value` function from the `text` widget module and replaced its usage with `Node::physical_size`.
* Derived `Copy` for `Node` (since it's only a wrapped `Vec2`).
* Made `Node::size` const.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Review by nicoburns — APPROVED** [review]

File: crates/bevy_ui/src/ui_node.rs [hunk 1]
```diff
@@ -30,6 +30,15 @@ impl Node {
         self.calculated_size
     }
 
+    /// Returns the size of the node in physical pixels based on the given scale factor.
+    #[inline]
+    pub fn physical_size(&self, scale_factor: f64) -> Vec2 {
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply crates/bevy_ui/src/ui_node.rs:35]: IIRC this is just a winit convention that we've stuck with.

[the remark this answers is not in the document]
Comment (ickshonpe) [reply crates/bevy_ui/src/ui_node.rs:35]: This hasn't seemed like a priority so I've left it alone but I can't see anywhere that it makes any difference. We're not dealing with large numbers, Taffy and bevy_text both do rounding of coordinates and almost everything is recalculated with each update. There's nowhere for errors to accumulate. 

If there are problems with just using `f32`s everywhere, I think they'd be very situational and we could just switch back in that specific situation. It makes no sense to use double precision when calculating things like mouse over and it's an extra little burden for devs and users when the scale factor stuff is already fragile and confusing.

**Comment by alice-i-cecile** [comment]
@ickshonpe can you resolve the merge conflicts and then I'll merge :)

**Comment by ickshonpe** [comment]
> @ickshonpe can you resolve the merge conflicts and then I'll merge :)

done

**Pull request merged**